src/metaopt/core/worker/experiment.py

```diff
@@ -154,10 +155,17 @@ def reserve_trial(self, score_handle=None):
         if not new_trials:
             return None
 
-        if score_handle is None:
-            selected_trial = random.sample(new_trials, 1)[0]
-        else:
-            raise NotImplementedError("scoring will be supported in the next iteration.")
+        if score_handle is not None and self.space:
+            scores = list(map(score_handle,
+                              map(lambda x: trial_to_tuple(x, self.space), new_trials)))
```

Comment:
Based on trial_to_tuple code, only hyper-parameter values are passed to the score_handle, not the results? The score_handle need to keep internally results information to match hyper-parameters with them?